File: src/MicrosoftTeamsIntegration.Jira/ClientApp/src/app/core/models/NotificationSubscription.ts
export interface NotificationSubscription {
    subscriptionId?: string;
    jiraId: string;
    subscriptionType: SubscriptionType;
    conversationId: string;
    conversationReference?: string;
    conversationReferenceId: string;
    eventTypes: string[];
    projectId: string;
    microsoftUserId: string;
    filter: string;
    isActive: boolean;
}

export enum SubscriptionType {
    Personal = 0,
    Channel = 1
}

export enum PersonalEventType {
    ActivityIssueCreator = 'ActivityIssueCreator',
    CommentIssueCreator = 'CommentIssueCreator',
    ActivityIssueAssignee = 'ActivityIssueAssignee',
    CommentIssueAssignee = 'CommentIssueAssignee',
    IssueViewer = 'IssueViewer',
    MentionedOnIssue = 'MentionedOnIssue',
    CommentViewer = 'CommentViewer'
}

export enum ChannelEventType {
    IssueCreated = 'IssueCreated',
    IssueUpdated = 'IssueUpdated',
    CommentCreated = 'CommentCreated',
    CommentUpdated = 'CommentUpdated'
}

File: src/MicrosoftTeamsIntegration.Jira/Models/NotificationSubscription.cs
using System;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson.Serialization.IdGenerators;

namespace MicrosoftTeamsIntegration.Jira.Models;

[Serializable]
public sealed class NotificationSubscription
{
    [BsonId(IdGenerator = typeof(StringObjectIdGenerator))]
    [BsonRepresentation(MongoDB.Bson.BsonType.ObjectId)]
    [BsonElement("subscriptionId")]
    public string SubscriptionId { get; set; }

    [BsonElement("jiraId")]
    public string JiraId { get; set; }

    [BsonElement("subscriptionType")]
    public SubscriptionType SubscriptionType { get; set; }

    [BsonElement("conversationId")]
    public string ConversationId { get; set; }

    [BsonElement("conversationReference")]
    public string ConversationReference { get; set; }

    [BsonIgnore]
    public string ConversationReferenceId { get; set; }

    [BsonElement("eventTypes")]
    public string[] EventTypes { get; set; }

    [BsonElement("projectId")]
    public string ProjectId { get; set; }

    [BsonElement("microsoftUserId")]
    public string MicrosoftUserId { get; set; }

    [BsonElement("filter")]
    public string Filter { get; set; }

    [BsonElement("isActive")]
    public bool IsActive { get; set; }
}

public enum SubscriptionType
{
    Personal = 0,
    Channel = 1
}

public enum PersonalEventType
{
    ActivityIssueCreator,
    CommentIssueCreator,
    ActivityIssueAssignee,
    CommentIssueAssignee,
    IssueViewer,
    MentionedOnIssue,
    CommentViewer
}

public enum ChannelEventType
{
    IssueCreated,
    IssueUpdated,
    CommentCreated,
    CommentUpdated
}
